Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -857,6 +857,44 @@ theorem constantCoeff_comp_algebraMap :
 
 end ConstantCoeff
 
+section coeffsIn
+variable {R S σ : Type*} [CommSemiring R] [CommSemiring S]
+
+section Module
+variable [Module R S] {M N : Submodule R S} {x : MvPolynomial σ S}
+
+variable (σ M)
+/-- The `R`-submodule of multivariate polynomials whose coefficients lie in a `R`-submodule `M`. -/
+@[simps]
+def coeffsIn : Submodule R (MvPolynomial σ S) where
+  carrier := {p | ∀ i, p.coeff i ∈ M}
+  add_mem' := by simp+contextual [add_mem]
+  zero_mem' := by simp
+  smul_mem' := by simp+contextual [Submodule.smul_mem]
+
+@[simp] lemma mem_coeffsIn : x ∈ coeffsIn σ M ↔ ∀ i, x.coeff i ∈ M := .rfl
+
+end Module
+
+section Algebra
+variable [Algebra R S] {M N : Submodule R S} {x : MvPolynomial σ S} {n : ℕ}
+
+lemma le_coeffsIn_mul : coeffsIn σ M * coeffsIn σ N ≤ coeffsIn σ (M * N) := by
```

Comment:
This can be upgraded to equality, right? You can get `m * n * X ^ k` by multiplying `m * X ^ 0` and `n * X ^ k`. The next one too.